java/HealthVault Mobile/HVSample/src/main/java/com/microsoft/hsg/android/hvsample/MainActivity.java

```diff
@@ -82,6 +86,38 @@ public void onClick(View view) {
 				}
 			}
 		});
+
+		LinearLayout profileTile = (LinearLayout) findViewById(R.id.profileTile);
+		profileTile.setOnClickListener(new View.OnClickListener(){
+			@Override
+			public void onClick(View view) {
+				Intent intent = null;
+				if (mService.isAppConnected()) {
+					intent = new Intent(MainActivity.this, ProfileActivity.class);
+				} else {
+					Toast.makeText(MainActivity.this, "Please connect to HV from Setting menu!", Toast.LENGTH_SHORT).show();
+				}
+				if(intent != null) {
```

Comment:
Instead of having this if we can just move the startActivity call up into line 96 where we assign a value to the intent and do it all in one shot. Then we can also get rid of the declaration of the intent variable